OAuth: Add optional strict parsing of role_attribute_path (#28021)
grafana/grafana: docs/sources/auth/generic-oauth.md

### Role mapping

> By default the user is assigned the role `Viewer` if `role_attribute_path` doesn't return a role for the user, this is not always desirable and can be disabled by setting `role_attribute_strict = true`, which will deny the user access if no role is returned.

Reviewer: ```suggestion
By default the user is assigned the role `Viewer` if `role_attribute_path` doesn't return a role for the user, this is not always desirable and can be disabled by setting `role_attribute_strict = true`, which will deny the user access if no role or an invalid role is returned.
```

I'm not reviewing the documentation, our docs squad will do that, but I think this might be worth mentioning 🙂

> By default the user is assigned the role `Viewer` if `role_attribute_path` doesn't return a role for the user, this is not always desirable and can be disabled by setting `role_attribute_strict = true`, which will deny the user access if no role or an invalid role is returned.

Reviewer 2: ```suggestion
By default, if  the`role_attribute_path` does not return a role, the user is assigned the `Viewer` role. You can disable the role assignment by setting `role_attribute_strict = true`. It denies user access if no role or an invalid role is returned.
```
